scipy/optimize/_differentiable_functions.py

```diff
@@ -207,65 +248,80 @@ def __init__(self, fun, x0, args, grad, hess, finite_diff_rel_step,
         self._lowest_x = None
         self._lowest_f = np.inf
 
+        # normalize workers
+        workers = workers or map
+
         finite_diff_options = {}
         if grad in FD_METHODS:
             finite_diff_options["method"] = grad
             finite_diff_options["rel_step"] = finite_diff_rel_step
             finite_diff_options["abs_step"] = epsilon
             finite_diff_options["bounds"] = finite_diff_bounds
+            finite_diff_options["workers"] = workers
+            finite_diff_options["full_output"] = True
         if hess in FD_METHODS:
             finite_diff_options["method"] = hess
             finite_diff_options["rel_step"] = finite_diff_rel_step
             finite_diff_options["abs_step"] = epsilon
             finite_diff_options["as_linear_operator"] = True
+            finite_diff_options["workers"] = workers
+            finite_diff_options["full_output"] = True
 
         # Initial function evaluation
+        self._nfev = 0
         self._update_fun()
 
         # Initial gradient evaluation
-        self._wrapped_grad, self._ngev = _wrapper_grad(
+        self._wrapped_grad = _GradWrapper(
             grad,
             fun=self._wrapped_fun,
             args=args,
-            finite_diff_options=finite_diff_options
+            finite_diff_options=finite_diff_options,
         )
         self._update_grad()
 
         # Hessian evaluation
-        if callable(hess):
-            self._wrapped_hess, self._nhev, self.H = _wrapper_hess(
-                hess, x0=x0, args=args
-            )
-            self.H_updated = True
-        elif hess in FD_METHODS:
-            self._wrapped_hess, self._nhev, self.H = _wrapper_hess(
-                hess,
-                grad=self._wrapped_grad,
-                x0=x0,
-                finite_diff_options=finite_diff_options
-            )
-            self._update_grad()
-            self.H = self._wrapped_hess(self.x, f0=self.g)
-            self.H_updated = True
-        elif isinstance(hess, HessianUpdateStrategy):
+        if isinstance(hess, HessianUpdateStrategy):
             self.H = hess
             self.H.initialize(self.n, 'hess')
             self.H_updated = True
             self.x_prev = None
             self.g_prev = None
-            self._nhev = [0]
+            _FakeCounter = namedtuple('_FakeCounter', ['ngev', 'nhev'])
```

Comment:
this `namedtuple` is created to allow `self._wrapped_hess` to have `ngev` and `nhev` attributes. In alternative code paths `self._wrapped_hess` is a `_HessWrapper` instance which has those attributes.